--- dwm.c	2021-12-15 17:41:59.190696565 +0530
+++ dwm.c	2021-12-15 17:02:43.160746933 +0530
@@ -1368,16 +1368,15 @@
 monocle(Monitor *m)
 {
 	unsigned int n = 0;
-    int oh, ov, ih, iv;
 	Client *c;

-    getgaps(m, &oh, &ov, &ih, &iv, &n);
-
-    if (n > 0) /* override layout symbol */
-        snprintf(m->ltsymbol, sizeof m->ltsymbol, "[%d]", n);
-
-    for (c = nexttiled(m->clients); c; c = nexttiled(c->next))
-        resize(c, m->wx + ov, m->wy + oh, m->ww - 2 * c->bw - 2 * ov, m->wh - 2 * c->bw - 2 * oh, 0);
+	for (c = m->clients; c; c = c->next)
+		if (ISVISIBLE(c))
+			n++;
+	if (n > 0) /* override layout symbol */
+		snprintf(m->ltsymbol, sizeof m->ltsymbol, "[%d]", n);
+	for (c = nexttiled(m->clients); c; c = nexttiled(c->next))
+		resize(c, m->wx, m->wy, m->ww - 2 * c->bw, m->wh - 2 * c->bw, 0);
 }

 void
